### DISCUSSION TITLE
VS Code setup help needed

### DISCUSSION THREAD

**Discussion by dbrans** [smartcontractkit/full-blockchain-solidity-course-py#283] [Q&A] [answered]
Hi folks,
I am really enjoying this course! Big thanks to Patrick and the community!

I am following along with the youtube video around [3h36m](https://www.youtube.com/watch?v=M576WGiDBdQ&t=12970s) and have a few things that are out of whack.

Setup: 
* I installed python 3.X using homebrew. `python3 --version` gives the correct version.
* I just installed VS Code along with Python and Solidity extensions. That is all.
* I am using [this version of SimpleStorage.sol](https://github.com/PatrickAlphaC/simple_storage/blob/c515e48/SimpleStorage.sol).

### Problem 1: Missing or strange settings.
When I search VS Code settings for "Solidity" I only see the following (image). There is no `"Solidity: Formatter"` for example. Also, the settings like `"compile using remote version"` must be edited using settings.json instead.

![image](https://user-images.githubusercontent.com/507015/139504999-6c3f4f63-5945-4dfb-97b3-7908c849db4c.png)

### Problem 2: pragma errors
I edited settings.json manually to edit the remote compiler:
![image](https://user-images.githubusercontent.com/507015/139505469-b7fe472d-de95-4e16-b272-bff504f51743.png)

However I get his error:
![image](https://user-images.githubusercontent.com/507015/139505406-0e20dd61-1be2-45d5-a161-7ee0eb0aa4d6.png)

... and this error:
![image](https://user-images.githubusercontent.com/507015/139505568-7ef14a33-68de-4bdb-9af9-f518c80afed6.png)

I appreciate everyone who took the time to read this and I hope my ignorance of VS Code (and blockchain for that matter) will be of help to others.

Regards,
Derek

**Comment by cromewar** [comment, marked as answer]
Hello @dbrans thanks your post your question here.

There is some extensions for solidity but the most recommended one is the extension by Juan Blanco

![image](https://user-images.githubusercontent.com/20199180/139507885-4fa9dc5c-73bf-462c-89d7-26c62cae2a48.png)

with this extension you easily can change the compiler version with a simple right click:

![image](https://user-images.githubusercontent.com/20199180/139507967-eebe1f4c-08e0-4433-bff5-6047ea43122a.png)

The option is: change global compiler version (remote)

with this extension you'll have now the option on the settings for the formatter:

![image](https://user-images.githubusercontent.com/20199180/139508067-1901a8e6-4017-453c-9939-6a2682788e70.png)

Also if you are on windows check the course index: Windows Support to know which other tools you would need for the course and the optimal configuration.

Feel free to make any other question, and welcome to blockchain world.

**Reply by dbrans** [reply]
I had installed a random Solidity plugin and not the intended Solidity plugin. Thank you.

**Reply by cromewar** [reply]
I'm glad it helped you.
Best regards, Crome.
